### DISCUSSION TITLE
Can you provide a better example converting from nodintegration = true to conTextBridge for Electron?

### DISCUSSION THREAD

**Discussion by delebash** [quasarframework/quasar#10702] [CLI - Electron mode]
It would be great if you could provide a complete conversion example.

Lets say I have in my render.vue
fs = require('fs')
store = require('electron-store')

Could you provide a complete example of what we need to put in the electron-preload.js, the App vue, Main vue, and Render vue in order to get it working like it was by setting nodeIntegration = true and contextIsolation = false. 

 Currently even if you set contextIsolation = false and nodeIntegration = true and run quasar dev -m electron it fails.  However in a standard electron app this still works as expected.  I know it it very insecure but if I you just local packages and have one api from a trusted site it would be nice to be able to use nodeIntegration = true.  But if you can give a nice complete example from the old nodeIntegration to the new way that would be great.  TY

**Comment by pikil** [comment]
I would really recommend following the secure way of setting the app:
https://quasar.dev/quasar-cli/developing-electron-apps/electron-preload-script#how-to-use-it

That approach works just fine - have been tested in a couple of applications.

**Reply by pikil** [reply]
And do not forget to setting that preload path in new BrowserWindow

```
...
    webPreferences: {
      contextIsolation: true,
      preload: path.resolve(__dirname, process.env.QUASAR_ELECTRON_PRELOAD)
    }
...
```

**Comment by delebash** [comment]
That is fine I have no problem with that.  It just that after searching the internet and looking through the examples it is still not very clear to me what is going on and how I convert from the old to the new contextBridge.  The example you listed does not really help me understand how to convert.  Any way you could provide a better example of how to convert from the example I stated above.  Maybe an example showing the old way and the steps you need to take to convert.

**Reply by pikil** [reply]
Sure, let's try this approach then :) Try the following example for electron store:

in electron-preload.js

```
const store = new ElectronStore()

contextBridge.exposeInMainWorld('electronStore', {
  set: (key, value) => {
    store.set(key, value)
  },
  get: key => store.get(key),
  delete: (key) => {
    store.delete(key)
  },
  clear: () => {
    store.clear()
  }
})
```

In *.vue or *.js file

```
this.storageHandler = (this.isElectron)
      ? window.electronStore
      : localStorage
```

**Reply by TianYa-Q** [reply]
I'm using electron-store in preload.js too. And I have a question about how to use cypress to test all electron app? thanks

**Reply by JokerMartini** [reply]
Yeah this new framework makes it impossible to use file watchers. Since you can't pass fs class objects from boot to store you can't actually listen to file changes. I wish someone would explain how to achieve this now that it no longer words like it used it.

**Reply by pikil** [reply]
Try to create a new discussion with the problematic code - that way someone would be able to help with the problem.

**Reply by JokerMartini** [reply]
I posted multiple times and asked around for help. I can post again and see. In my case I would just like to see a simple example on how to create an electron app that supports FileWatchers. This does not seem possible any more with V2 and Vue3.

**Reply by pikil** [reply]
Can you post those links here?

**Reply by JokerMartini** [reply]
https://github.com/quasarframework/quasar/discussions/10012
